### ISSUE TITLE
Tailscaled not starting

### ISSUE THREAD

**Issue by davemecha** [gitpod-io/gitpod#7142]
### Bug description

When starting a workspace with tailscale (e.g. the tailscale template) the tailscale daemon does not start.

Here is the console output from starting `tailscaled`:

```
logtail started
Program starting: v1.18.1-t0a4334048-gb05dc05d4, Go 1.17.2-ts7037d3ea51: []string{"tailscaled"}
LogID: c1dad10085f8d24c622790caacfea153a5bde58d7f6e6ff8d30074e2a86529f4
logpolicy: using system state directory "/var/lib/tailscale"
logpolicy.Read /var/lib/tailscale/tailscaled.log.conf: open /var/lib/tailscale/tailscaled.log.conf: no such file or directory
wgengine.NewUserspaceEngine(tun "tailscale0") ...
Linux kernel version: 5.13.0-1007-gcp
is CONFIG_TUN enabled in your kernel? `modprobe tun` failed with: 
tun module not loaded nor found on disk
wgengine.NewUserspaceEngine(tun "tailscale0") error: tstun.New("tailscale0"): operation not permitted
wgengine.New: tstun.New("tailscale0"): operation not permitted
flushing log.
logger closing down
logtail: dialed "log.tailscale.io:443" in 165ms
``` 

### Steps to reproduce

* Add a `TAILSCALE_AUTHKEY` to [Gitpod environment variables](https://gitpod.io/variables) as described in the [docs](https://www.gitpod.io/docs/configure/tailscale)
* Launch the [template-tailscale](https://github.com/gitpod-io/template-tailscale) workspace
* Check the `tailscaled` terminal for log output

### Expected behavior

The daemon should start so I can connect to the tailscale network.

### Example repository

https://github.com/gitpod-io/template-tailscale

**Comment by csweichel** [comment]
This does not come from a recent tailscale change but is most likely related to an infrastructure change on our end.

**Comment by csweichel** [comment]
Quote from Discord:
> Tailscale somehow stopped working today. Last night the connection was still working but after a restart of the workspace (and also after creating a new one) the tailscaled does not start.

Working on 08.12.2021
Broken on 09.12.2021

**Comment by davemecha** [comment]
I restarted my container yesterday (08.12.) several times and it worked everytime. When I started the container today in the morning, there was a short build screen when opening the container. Some new builds were done. After the container started, I had the tailscale error.

**Comment by ghuntley** [comment]
Confirming that https://ghuntley.com/new is now broken.

<img width="870" alt="Screen Shot 2021-12-09 at 11 23 25 pm" src="https://user-images.githubusercontent.com/127353/145404328-cf1b7a41-0fd2-49a2-8548-fef6734d6129.png">

**Comment by csweichel** [comment]
tailscaled v1.16.2 exhibits the same problem. This is not related to the tailscale version

**Comment by axonasif** [comment]
A few days back after gitpod moving to GCP, I logged the kernel version to be `5.13.0-1006-gcp` but today it seems to be `5.13.0-1007-gcp`, maybe this kernel build got that module missing in the host machine or maybe it's not loaded there?


As per the error logs:
```
Linux kernel version: 5.13.0-1007-gcp
is CONFIG_TUN enabled in your kernel? `modprobe tun` failed with: 
tun module not loaded nor found on disk
```
`tun` is not present under `/sys/module/`, I had tried to load it manually via installing `5.13.0-1007-gcp` kernel modules but seems like modprobing has no effect under gitpod workspaces sadly. I had tried the following in a gitpod workspace:
```bash
echo 'deb https://archive.ubuntu.com/ubuntu/ impish-updates main' | sudo tee -a /etc/apt/sources.list
sudo apt update
sudo apt install -y linux-base linux-modules-5.13.0-1007-gcp
sudo modprobe tun && ls /sys/module/tun # No such dir
```

**Comment by csweichel** [comment]
I reckon the kernel version change you noticed comes from when we moved from the ws20 to the ws21 clusters. I checked the VM images we're running and they've remained stable (as expected).

The syscall that fails now is:
```
ioctl(11, TUNSETIFF, 0xc0000ab970) = -1 EPERM (Operation not permitted)
```
i.e. all of a sudden, from within workspaces we can no longer set the name of the tun device we get from `/dev/tap/tun`. We do get a device which means it’s not a missing `tun` module in the kernel.

Maybe we’ve hit some kind of limit on the nodes. However, we have new and old nodes and I have yet to find a workspace which would not be affected.

**Comment by axonasif** [comment]
So it seems that tun is a built-in module.

**Comment by davemecha** [comment]
In case that's helpful: My workspace that stopped working from yesterday to today was on `ws-eu21.gitpod.io` and it was AFAIK not moved to another host between restarts.

**Comment by csweichel** [comment]
We have found the issue:
- two days ago we made the "experimental network mode" the default (#7063). As part of that change we removed the "WORKSPACEKIT_USE_NETNS" environment variable
- this morning we deployed this change, but not the change to workspacekit which enabled the "experimental network mode" by default
- at this point the experimental network mode code no longer be enabled, because `server` would not set the environment variable anymore, and `workspacekit` would still require it.

How we'll fix this:
- we'll deploy a version of `workspacekit` which exhibits the network behaviour we had intended

What went wrong while debugging:
- we tried this on new workspace clusters of which we believed they came with a new workspacekit version. It took a while until we questioned that, and in the process found that workspacekit is delivered as part of the supervisor image, which is independent of the workspace cluster version. This led us to falsely believe that it must be an infrastructure change of some sort.
- we spent too much time looking into low-level stuff (e.g. all the way [until here](https://elixir.bootlin.com/linux/v5.13/source/drivers/net/tun.c#L2670)), because we believed that something in the cluster must have changed

What does this mean for you?
We've brought a fix online which enables the desired networking behaviour. With this, tailscale should "just work again".
If your workspace carries a `ws-*23.gitpod.io` in the URL, things should be ok.
